nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -435,6 +436,9 @@ impl<'a> Planner<'a> {
             DiscretionaryOmicronZone::InternalDns => {
                 self.input.target_internal_dns_zone_count()
             }
+            DiscretionaryOmicronZone::ExternalDns => {
+                self.blueprint.count_parent_external_dns_zones()
```

Comment:
This will need an update when we implement cleanup of expunged external DNS zones, right?  I'm trying to think of how to make sure we notice this when it comes to that ... (CC @sunshowers)

Reply:
I thought about that some too, but did not get any farther than "we'll probably need a special case". It may or may not be a good reason to increase the relative priority of #3732.

Reply:
Should be fixed by 8fe56d513, which collects a unique set of external DNS IP addresses over all zones and counts those.